### ISSUE TITLE
SO3 sixd -> rotmat diverges from common Kimodo/PyTorch 6D reconstruction on noisy network outputs

### ISSUE THREAD

**Issue by abcamiletto** [abcamiletto/nanomanifold#32]
While simplifying a downstream motion model port, I compared several `nanomanifold` SO(3) conversions against a saved upstream reference and isolated one of the larger end-to-end drifts to `SO3.convert(..., src="sixd", dst="rotmat")`.

Context:
- Downstream task: Kimodo text/constraint-conditioned motion generation
- Framework: PyTorch
- Comparison: end-to-end output tensors against a saved upstream reference
- Metric: `max_abs_diff`

Observed behavior when swapping only `sixd -> rotmat` into the pipeline:
- worst case: `1.1993981897830963e-03`
- worst field/case: `unconditional / local_rot_mats`
- other cases: roughly `1.7e-6` to `4.1e-6`

Control measurements from the same run:
- `matrix_to_cont6d` alone: `0.0`
- `axis_angle_to_matrix` alone: `0.0`
- `quaternion_to_matrix` alone: `<= 5.96e-7`

So this seems specifically tied to `from_sixd_to_rotmat`.

Current `nanomanifold` implementation in `nanomanifold/SO3/primitives/sixd.py`:
```python
b1 = normalize(a1)
dot = sum(b1 * a2)
b2 = normalize(a2 - dot * b1)
b3 = cross(b1, b2)
return stack([b1, b2, b3], axis=-1)
```

Reference implementation from the downstream codebase:
```python
def _cont6d_to_matrix(cont6d: Tensor) -> Tensor:
    if cont6d.shape[-1] != 6:
        raise ValueError(f"The last dimension must be 6, got {cont6d.shape[-1]}.")
    x = cont6d[..., 0:3]
    y = cont6d[..., 3:6]
    x = x / torch.norm(x, dim=-1, keepdim=True)
    z = torch.cross(x, y, dim=-1)
    z = z / torch.norm(z, dim=-1, keepdim=True)
    y = torch.cross(z, x, dim=-1)
    return torch.stack([x, y, z], dim=-1)
```

The important differences are:
- `nanomanifold` orthogonalizes the second vector with `a2 - dot * b1` before normalizing it.
- the downstream implementation constructs the third axis first with `cross(x, y)` and then recomputes the second axis as `cross(z, x)`.
- both are valid SO(3) projections, but they are not numerically equivalent on imperfect network outputs.

In this particular model, that difference is large enough to show up in the final motion.

Feature request:
- could `nanomanifold` expose an alternate `sixd -> rotmat` mode matching the cross-product reconstruction above?
- alternatively, if the current behavior is intentional/preferred, it would help to document that it may differ from the reconstruction used by some motion-model codebases.

I can provide a minimal downstream repro or frozen tensors if helpful.
